### ISSUE TITLE
[Bug] Structure compass will search past the configured limit, causing most of the server to time out from a lag spike.

### ISSUE THREAD

**Issue by JustAnotherGhostName** [AllTheMods/ATM-8#682]
The structure compass searches past the configured limit which causes most players on a server to time out. This notably occurs with the allthemodium:ancient_pyramid structure, though I am assuming it occurs with other structures as well since people only use the compass for the pyramid.

This issue was also reported to the Structure Compass GitHub repo in the following bug report:
https://github.com/Mrbysco/StructureCompass/issues/9 

Currently the server is running on ATM8 1.0.7. As the structure compass mod did not get updated in 1.0.8, I am assuming the issue still occurs on there as well.

**Comment by joakime** [comment]
The configured `compassRange` in ATM8 is 5000. that's 5000 chunks.

The default that comes with StructureCompass is 300.

Maybe this configuration in ATM8 should be reverted for sanity sake

**Comment by JustAnotherGhostName** [comment]
> The configured `compassRange` in ATM8 is 5000. that's 5000 chunks.
> 
> The default that comes with StructureCompass is 300.
> 
> Maybe this configuration in ATM8 should be reverted for sanity sake

This was not the cause, though that did help. Mod author adjusted to also be in blocks and did a rework in 1.4.0. I updated to it after he changed it, and issue still occurred. I then isolated the issue to the default value (by atm8 im pretty sure, i dont recall changing it) for looking for unexplored structures. That specific tag caused massive spikes.
